types/codespacer.go

```diff
@@ -0,0 +1,39 @@
+package types
+
+// Codespacer is a simple struct to track reserved codespaces
+type Codespacer struct {
+	next     CodespaceType
+	reserved map[CodespaceType]bool
+}
+
+// NewCodespacer generates a new Codespacer with the starting codespace
+func NewCodespacer() *Codespacer {
+	return &Codespacer{
+		next:     CodespaceType(1),
+		reserved: make(map[CodespaceType]bool),
+	}
+}
+
+// reserve reserves a specified codespace
+func (c *Codespacer) reserve(codespace CodespaceType) {
+	if codespace == c.next {
+		c.next++
```

Comment:
buggy.  if/else-if/else is skipped after .next++ and no reservation takes place.  This should return a bool or register the next available after codespace, and return that codespace.

Reply:
Yes, this was silly, simplified